key.core/src/main/antlr4/KeYParser.g4

```diff
@@ -99,7 +99,7 @@ one_sort_decl
 
 simple_ident_dots
 :
-  simple_ident (DOT simple_ident)* | INT_LITERAL
```

Comment:
For some reason, a number (1,2,3, etc) can also be treated as a dotted identifier (e.g., `java.lang.Object`). 

Hence, `2522` would be a valid name of a sort (`sordId` derives to `simple_ident_dots`). 

This lead to an ambiguity: `23423` can be an `integer`, or and `accessterm`.

Reply:
I think that 0,1,2,...,9 are used as function names and function names are `simple_ident_dots`. This might be solved by moving the `INT_LITERAL` to `funcpred_name` if that was the reason.

Reply:
I find the fact that `1` can be an int constant (resolving to `Z(1(#))`) and at the same time the name of the function used inside the constants a little unsatisfying, I must admit. (But ... is it worth changing that?)

@unp1

Reply:
Couldn't they just be called `ZN` or `_N`?

Reply:
If you mean `Z0, Z1, ...` then I would say this would be clearer; but a change at the very roots.

Reply:
On the other side, we have a special lexer thing, that lifts numbers before parens to identifier. But I needed to add `INT_LITERAL` to `funcpred_name` to parse some files again.

https://github.com/KeYProject/key/blob/a9ec83b0c80fa10bdf24dc9b94713c19d45d18d5/key.core/src/main/antlr4/KeYLexer.g4

```
   public void emit(Token token) {
       int MAX_K = 10;
       if (token.getType() == INT_LITERAL) {//rewrite INT_LITERALs to identifier when preceeded by an '('
           for (int k = 1; k <= MAX_K; k++) {
               int codePoint = _input.LA(k);
               if (Character.isWhitespace(codePoint)) continue;
               if (codePoint == '(') ((org.antlr.v4.runtime.CommonToken) token).setType(IDENT);
               break;
           }
       }
       if(token.getType() == PROOF) {
          tokenBackStorage = super.emitEOF();
          //will later be overwritten the EOF token
       }
       super.emit(token);
    }
```